src/LMS1xx.cpp

```diff
@@ -187,6 +187,18 @@ void LMS1xx::setScanDataCfg(const scanDataCfg &cfg) {
 	buf[len - 1] = 0;
 }
 
+void LMS1xx::setOutputRange(const scanOutputRange &cfg) {
+	char buf[100];
+	sprintf(buf, "%c%s 1 %X %X %X%c", 0x02, "sWN LMPoutputRange",
+			cfg.angleResolution, cfg.startAngle, cfg.stopAngle, 0x03);
+
+	write(sockDesc, buf, strlen(buf));
```

Comment:
`strlen` would return the length until the first null character `0x00`. See http://www.cplusplus.com/reference/cstring/strlen/

Since `buf[100]` isn't initialized to anything, the return value is unexpected. You can solve that adding `0x00` after `0x03`.

However, if any of the `cfg` fields introduces a `0x00` byte into the `buf` array, the `strlen` would be smaller than it should.

I think it'd be better to take the length from the return value of `sprintf`. See http://www.cplusplus.com/reference/cstdio/sprintf/